src/components/SpiceItem.js

```diff
@@ -3,6 +3,21 @@ import StarRating from './StarRating'
 
 class SpiceItem extends React.Component {
 
+  state = {
+    filledHeart: false
+  }
+
+  toggleHeart = evt => {
+    const callback = prevState => {
+      return {
+        filledHeart: !prevState.filledHeart
+      }
+    }
+
+    this.setState(callback)
```

Comment:
Nice work with the callback version of setState here! You typically will see the callback defined inside setState directly instead of saved to a variable, but there's nothing wrong with your approach. Just to show you the other syntax:

```js
this.setState(prevState => {
  return {
    filledHeart: !prevState.filledHeart
  }
})
```